src/vm_pages.rs

```diff
@@ -1264,6 +1319,75 @@ impl<'a, T: GuestStagePagingMode> FinalizedVmPages<'a, T> {
         self.do_remove_region(page_addr, len, VmRegionType::Mmio)
     }
 
+    /// Converts the specified memory region from confidential to shared. Returns the TLB version
+    /// at which the conversion will be completed.
+    pub fn share_mem_region(&self, page_addr: GuestPageAddr, len: u64) -> Result<TlbVersion> {
+        let end = PageAddr::new(
+            RawAddr::from(page_addr)
+                .checked_increment(len)
+                .ok_or(Error::AddressOverflow)?,
+        )
+        .ok_or(Error::UnalignedAddress)?;
+        // Check the address range lies within a "Confidential" region of guest physical address
+        // space and update the region type to "ConfidentialRemovable".
+        self.inner.regions.write().update_in_place(
+            page_addr,
+            end,
+            VmRegionType::Confidential,
+            VmRegionType::ConfidentialRemovable,
+            false,
+        )?;
+
+        // Should we go further here, by getting the actual list of pages that will be
+        // invalidated, so that we can know for sure if we can return to the guest.
+        Ok(self.inner.tlb_tracker.current_version().increment())
```

Comment:
I kept incrementing the TLB (following what was already there in `share_mem_region_begin()`) because that's the way to tell the calling function `handle_tee_guest_msg()` that we have a blocking ecall. But I guess with the new design we can always assume a `ShareMemory` or `UnshareMemory` call is a blocking ecall. And that would allow to get rid of this logic here.